app/code/Magento/Wishlist/Model/Wishlist.php

```diff
@@ -3,32 +3,30 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
 namespace Magento\Wishlist\Model;
 
 use Magento\Catalog\Api\ProductRepositoryInterface;
 use Magento\Framework\App\ObjectManager;
+use Magento\Framework\DataObject\IdentityInterface;
 use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Model\AbstractModel;
 use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Wishlist\Api\Data\WishlistInterface;
 use Magento\Wishlist\Model\ResourceModel\Item\CollectionFactory;
 use Magento\Wishlist\Model\ResourceModel\Wishlist as ResourceWishlist;
 use Magento\Wishlist\Model\ResourceModel\Wishlist\Collection;
 
 /**
  * Wishlist model
  *
- * @method int getShared()
- * @method \Magento\Wishlist\Model\Wishlist setShared(int $value)
- * @method string getSharingCode()
- * @method \Magento\Wishlist\Model\Wishlist setSharingCode(string $value)
- * @method string getUpdatedAt()
- * @method \Magento\Wishlist\Model\Wishlist setUpdatedAt(string $value)
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  * @SuppressWarnings(PHPMD.TooManyFields)
  *
  * @api
  * @since 100.0.2
  */
-class Wishlist extends \Magento\Framework\Model\AbstractModel implements \Magento\Framework\DataObject\IdentityInterface
+class Wishlist extends AbstractModel implements IdentityInterface, WishlistInterface
```

Comment:
This class seems to be a legacy model, and trying to implement `WhishlistInterface` on top of it may be one of the less-optimal solutions. 

I would suggest implementing a separate data transfer object to implement the interface.

Reply:
I've created a new dto in Model\Data\Wishlist which now implements the interface, wishlist now has public function getDataModel which copies properties to new implementation and returns. Repository has been changed to directly return new dto instead of legacy one. All of this has been based on implementation in Customer files Model\Customer, Model\Data\Customer and CustomerRepository.